### ISSUE TITLE
Roslyn incorrectly processes `NotNullIfNotNull` attribute which referenes `params` parameter if there are no corresponding arguments

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#37832]
**Version Used**: 

```
Date:   Mon Aug 5 16:00:23 2019 -0700

    Merge pull request #37730 from dotnet/merges/release/dev16.3-preview2-to-master

    Merge release/dev16.3-preview2 to master
```

**Steps to Reproduce**:

Compile the following code:
```cs
using System.Diagnostics.CodeAnalysis;

public class C
{
    [return: NotNullIfNotNull("p")]
    string? M(params string?[]? p) => p?.ToString();
   
    void Test1()
    {
        var s = M();
        s.ToString();
    }
}
```

**Expected Behavior**:
No warnings. An empty array will be passed to `p`.

**Actual Behavior**:
`warning CS8602: Dereference of a possibly null reference.` is reported for `s.ToString()`

**Comment by RikkiGibson** [comment]
This bug is still present. I found the warning behavior somewhat interesting:

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;

public class C {
    [return: NotNullIfNotNull("arr")]
    public static string?[]? M(params string?[]? arr) {
        return arr;
    }
    
    public static void Main() {
        M(new string[] { }).ToString(); // no warning
        M().ToString(); // warning
        
        M("a").ToString(); // no warning
        M((string?)null).ToString(); // no warning
    }
}
```

**Comment by RikkiGibson** [comment]
This might be a case which can be solved by #49602, which removes the need for flow analysis, etc. to "translate" the arguments for a params array parameter.
